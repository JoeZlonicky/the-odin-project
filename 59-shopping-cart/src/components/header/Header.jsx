import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import CartCount from '../cart-count/CartCount';
import styles from './Header.module.css';

const Header = ({ cart }) => {
  return (
    <header className={styles.header}>
      <div className={styles.content}>
        <div className={styles.titleAndAuthor}>
          <NavLink to="/" className={styles.title}>
            The Tree Store
          </NavLink>
          <div className={styles.author}>By Joe Zlonicky</div>
        </div>

        <nav className={styles.nav}>
          <NavLink to="/" className={({ isActive }) => (isActive ? styles.active : '')}>
            Store
          </NavLink>
          <NavLink to="/cart" className={({ isActive }) => (isActive ? styles.active : '')}>
            Cart <CartCount cart={cart} />
          </NavLink>
        </nav>
      </div>
    </header>
  );
};

Header.propTypes = {
  cart: PropTypes.instanceOf(Map),
};

export default Header;
