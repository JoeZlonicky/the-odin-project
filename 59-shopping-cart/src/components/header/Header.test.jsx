import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { expect, it } from 'vitest';
import Header from './Header';

it('Matches screenshot', () => {
  const { container } = render(
    <MemoryRouter>
      <Header cart={new Map()} />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});

it('Correctly navigates to home', async () => {
  const user = userEvent.setup();

  render(
    <MemoryRouter initialEntries={['/cart']}>
      <Header cart={new Map()} />
    </MemoryRouter>,
  );
  await user.click(screen.getByText('Home'));

  const activeLinks = screen.getAllByRole('link', { current: 'page' });
  expect(activeLinks[0].textContent).toMatch('PokÃ© Mart');
  expect(activeLinks[1].textContent).toMatch('Home');
});

it('Correctly navigates to all products', async () => {
  const user = userEvent.setup();

  render(
    <MemoryRouter initialEntries={['/']}>
      <Header cart={new Map()} />
    </MemoryRouter>,
  );
  await user.click(screen.getByText('All Products'));

  const activeLink = screen.getByRole('link', { current: 'page' });
  expect(activeLink.textContent).toMatch('All Products');
});

it('Correctly navigates cart', async () => {
  const user = userEvent.setup();

  render(
    <MemoryRouter initialEntries={['/']}>
      <Header cart={new Map()} />
    </MemoryRouter>,
  );
  await user.click(screen.getByText(/Cart/));

  const activeLink = screen.getByRole('link', { current: 'page' });
  expect(activeLink.textContent).toMatch(/Cart/);
});
