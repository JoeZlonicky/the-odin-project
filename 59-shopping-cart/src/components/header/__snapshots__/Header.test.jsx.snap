// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches screenshot 1`] = `
<div>
  <header
    class="_header_69e643"
  >
    <div
      class="_content_69e643"
    >
      <div
        class="_titleAndAuthor_69e643"
      >
        <a
          aria-current="page"
          class="_title_69e643 active"
          href="/"
        >
          Pok√© Mart
        </a>
        <div
          class="_author_69e643"
        >
          By Joe Zlonicky
        </div>
      </div>
      <nav
        class="_nav_69e643"
      >
        <a
          aria-current="page"
          class="_active_69e643"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/all-products"
        >
          All Products
        </a>
        <a
          class=""
          href="/cart"
        >
          Cart (
          0
          )
        </a>
      </nav>
    </div>
  </header>
</div>
`;
